### ISSUE TITLE
Feature: Hitting back during "keypad" input goes back to previous group of characters

### ISSUE THREAD

**Issue by hojdan** [digitalbitbox/bitbox02-firmware#862]
Let's say I want to enter the letter a as a simple example.

1. I touch the group with a-h
2. I touch the next group that appears which is a-c
3. I then touch a

Let's say I click something wrong by accident at some point.

1. If I touch the next group i-q, touching the back button afterward will take me back to all the letters and then I can touch the correct group. No issue here.
2. If I touch the next group d-e, touching the back button afterward will take me all the way back to all the letters. Then I would need to first touch group a-h from all the letters then I can touch the correct group. I would prefer that touching the back button not take me back all the way but instead to just the previous group selection.
3. I click b by accident and thus b is entered. To correct my mistake I click the back button. No issue here.

**Comment by benma** [comment]
Makes sense to me. @jadzeidan what do you think?

**Comment by jadzeidan** [comment]
Yeah, also makes sense to me to go back to previous group. Ack from me.

**Comment by c266** [comment]
Just wanted to bump this, would be a great improvement.

**Comment by alessandro-opensource** [comment]
Hi! As mentioned in the PR I opened: https://github.com/BitBoxSwiss/bitbox02-firmware/pull/1689#issue-3631273880

I built it in the container and tested it with the simulator, and it seems to behave as expected.

Would love any feedback!
